MDANSE/Src/MDANSE/Framework/Configurators/IConfigurator.py

```diff
@@ -337,16 +342,14 @@ def check_dependencies(self, configured: list[str] | None = None) -> bool:
 
     def preview_output_axis(self):
         """Show what data axis will be created in the output file."""
-        if not self.show_prediction or not self.is_configured() or not self.valid:
-            yield None
-        elif hasattr(self, "prediction_keys"):
+        if not self.is_configured() or not self.valid:
+            return
+        if hasattr(self, "prediction_keys"):
             for key in self.prediction_keys:
-                yield (key, self[key], self.prediction_unit)
-        elif self.prediction_key in self:
+                yield (key, self[key], self.prediction.unit)
+        elif self.prediction.key in self:
             yield (
-                self.prediction_label,
-                self[self.prediction_key],
-                self.prediction_unit,
+                self.prediction.label,
+                self[self.prediction.key],
+                self.prediction.unit,
             )
```

Comment:
```suggestion
            yield self.prediction
```
This is already a tuple-like entity.

Reply:
`yield self.prediction` will give us ("label", "key", "unit"). The current version gives ("label", Sequence[float], "unit"). The `key` is used to pick one of the many dictionary entries in the `Configurator` which contains numbers we can use.

Reply:
Ah, I see, `key` is a key in the `IJob` dictionary. A little confusing, might be worth a quick comment.